time.go

```diff
@@ -8,6 +8,10 @@ import (
 type Time time.Time
 
 func (t *Time) UnmarshalJSON(b []byte) error {
+	if len(b) == 0 {
```

Comment:
I think this should rather error if len(b) == 0.

Reply:
I thought unmarshal/marshal should work end to end. Depends on what marshalling returns I assume. If it is not required, then happy to convert.

Reply:
If we used JSON marshalling for this use case, a user of the SDK who wanted to marshal a Time for other purposes would lose data when unmarshalling (we through away sub-millisecond precision).